feature_details.md

```diff
@@ -0,0 +1,417 @@
+
+* The `handleTransition` event has a field `previousURL` : the URL for the previous Document. And `previousElements` : the list of placeholders for elements passed to `setSameOriginTransitionElements`, clarified in [API Proposal](#api-proposal).
+
+* The `startTransition` API on `handleTransition` event takes a promise to allow script to asynchronously prepare the new Document for first paint. The browser continues to display old content until the promise passed to this API resolves.
+
+* The animations executed on root elements for both Documents during the transition are based on the `transitionType` specified for each.
+
+### Single Element Transition
+We can also set a separate animation for parts of the page by specifying them separately in the API. Let's say we wanted the header to slide up during this transition :
+
+TODO : Add rough implementation for video.
+
+Add the elements which will animate independently in the API call on the previous Document. The result is that the browser caches a separate pixel copy for each element in the list.
+
+```
+document.documentTransition.setSameOriginTransitionElements(
+  {
+   “foo.com/a.html” : [{element : document.html},
+                       {element : document.getElementById("header")}],
+  });
+```
+
+The `handleTransition` event on the new Document specifies the animation for the header as follows :
+
+```
+document.documentTransition.addEventListener(
+    “handleTransition”, e => {
+        e.startTransition(transitionFrom(e.previousUrl, e.previousElements));
+    });
+
+async function transitionFrom(previousUrl, previousElements) {
+    ...
+    
+    let headerAnimation = 
+      {previousElement : previousElements[1],
+       transitionType : "slide-up"};
+       
+    return [oldRootAnimation, newRootAnimation, headerAnimation];
+}
+```
+
+### Paired Element Transition
+In some cases a semantically or visually same element is present in both Documents. We can set up an animation which automatically animates this element from it's old state to the new state :
+
+[![Video Link for Shared Element Transition](https://img.youtube.com/vi/GekohitaU5M/0.jpg)](https://www.youtube.com/watch?v=GekohitaU5M)
+
+Similar to the case above, update the list in `setSameOriginTransitionElements` to include the animated image :
+
+```
+document.documentTransition.setSameOriginTransitionElements(
+  {
+   “foo.com/a.html” : [{element : document.html},
+                       {element : document.getElementById("dog")}],
+  });
+```
+
+And specify a transition which associates these elements in the `handleTransition` event on the new Document :
+
+```
+document.documentTransition.addEventListener(
+    “handleTransition”, e => {
+        e.startTransition(transitionFrom(e.previousUrl, e.previousElements));
+    });
+
+async function transitionFrom(previousUrl, previousElements) {
```

Comment:
`provideTransitionsFrom` would it make it clearer that this method returns transitions.